modules/ai/assets/js/editor/hooks/use-animation-prompt.js

```diff
@@ -0,0 +1,203 @@
+			isParent: true,
+			label: getControlData( firstEnableControlKey ).label,
+			value: jsonObject.enabled[ firstEnableControlKey ],
+			tabs: 0,
+		};
+
+		// Process other properties
+		const processProps = ( props ) => {
+			for ( const [ key, value ] of Object.entries( props ) ) {
+				if ( 'enabled' === key && value ) {
+					const enableControlKey = Object.keys( value )[ 0 ];
+					result[ enableControlKey ] = {
+						isParent: false,
+						label: getControlData( enableControlKey ).label,
+						value: value[ enableControlKey ],
+						tabs: 1,
+					};
+				} else {
+					if ( 'none' === value || 'normal' === value ) {
+						continue;
+					}
+					result[ key ] = {
+						value,
+					};
+				}
+			}
+		};
+
+		// Process all other inner properties
+		for ( const [ propKey, propValue ] of Object.entries( jsonObject ) ) {
+			if ( propKey !== 'enabled' && 'object' === typeof propValue ) {
+				processProps( propValue );
+			}
+		}
+
+		return result;
+	};
+
+	const processEntranceEffects = ( jsonObject ) => {
+		const result = {};
+
+		const firstChildKey = Object.keys( jsonObject )[ 0 ];
+		const firstChild = jsonObject[ firstChildKey ];
+
+		result[ firstChildKey ] = {
+			isParent: true,
+			label: getControlData( firstChildKey ).label,
+			value: firstChild,
+			tabs: 0,
+		};
+
+		result[ firstChild ] = {
+			isParent: false,
+			label: firstChild,
+			tabs: 1,
+		};
+
+		const processProps = ( props ) => {
+			for ( const [ key, value ] of Object.entries( props ) ) {
+				if ( 'none' === value || 'normal' === value ) {
+					continue;
+				}
+				result[ key ] = {
+					value,
+				};
+			}
+		};
+
+		for ( const [ propKey, propValue ] of Object.entries( jsonObject ) ) {
+			if ( propKey !== firstChildKey && 'object' === typeof propValue ) {
+				processProps( propValue );
+			}
+		}
+
+		return result;
+	};
+
+	const processHoverEffects = ( jsonObject ) => {
+		const result = {};
+		let isHoverLabelAdded = false;
+		const processProps = ( props ) => {
+			for ( const [ key, value ] of Object.entries( props ) ) {
+				if ( 'enabled' === key && value ) {
+					const enableControlKey = Object.keys( value )[ 0 ];
+					const { label, innerTab } = getControlData( enableControlKey );
+
+					if ( ! isHoverLabelAdded ) {
+						result[ innerTab ] = {
+							isParent: true,
+							label: getControlData( innerTab ).label,
+							tabs: 0,
+						};
+						isHoverLabelAdded = true;
+					}
+
+					result[ enableControlKey ] = {
+						isParent: false,
+						label,
+						value: value[ enableControlKey ],
+						tabs: 1,
+					};
+				} else {
+					if ( 'none' === value || 'normal' === value ) {
+						continue;
+					}
+					result[ key ] = {
+						value,
+					};
+				}
+			}
+		};
+
+		for ( const [ , propValue ] of Object.entries( jsonObject ) ) {
+			if ( 'object' === typeof propValue ) {
+				processProps( propValue );
+			}
+		}
+
+		return result;
+	};
+
+	const digestResult = ( jsonString ) => {
```

Comment:
this whole file and this part in particular is pretty complex, how about some unit tests to test the logic?